_posts/2018-06-15-cocoapods-keys-react-native.markdown

```diff
@@ -0,0 +1,156 @@
+---
+layout: epic
+title: Managing Secrets in a React Native App
+date: 2018-06-15
+author: erik
+categories: [Technology, emission, reactnative, react, javascript, ios]
+---
+
+<!--
+Iɴᴛʀᴏᴅᴜᴄᴛɪᴏɴ
+-->
+
+Hi! I'm Erik, a software engineer on the Purchase team. One of the most visible payoffs from Artsy's investments in React Native [over the past two years][react-native-tag] has been the opening up of our mobile codebase to contributors like myself coming primarily from web stacks. It's nice to be able to build mobile interfaces with the same declaritive API used by so many of our web projects, but sometimes we still need to bridge the divide to our Objective-C and Swift ecosystem. One such case: replacing the app secrets typically loaded from a deploy environment or web developer's [dotenv][] file.
+
+<!-- more -->
+<!--
+Mᴀɪɴ Bᴏᴅʏ
+-->
+
+[Emission][] is Artsy's React Native component library. It contains its own native app in the `/Example` folder, which uses [cocoapods-keys][] to store secrets while still letting us code in the open. In order to expose these keys to our react-native components we must do a fair bit of setup. To be frank: When I first paired with [@ash][ash] on this change I felt like I was driving blindfolded. It was only 2 days later that I revisited our PR to document the process for future developers and the pieces fell into place. It's a straightforward how-to that also makes a quick tour through the iOS ecosystem for web developers looking at React Native today.
+
+Links to examples below come from [this commit](https://github.com/artsy/emission/pull/1086/commits/4a2a3e9260e97d791536cf38376a06b0ad0946a8) which adds a key for the Stripe API to Emission. We'll be adding this key to an `Emission` module that can be imported to our React Native files.
+
+## Steps
+
+#### 1. Add the key to the app's Podfile.
+
+This is the extent of `cocoapods-keys` official [setup][ck-setup], and after this you **could** set the key via `pod keys set <NAME>` or `pod install`... but we have more to do.
+
+[/Example/Podfile](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Example/Podfile#L63):
+
+```diff
+plugin 'cocoapods-keys', {
+  :target => 'Emission',
+  :keys => [
+    'ArtsyAPIClientSecret',   # Authing to the Artsy API
+    'ArtsyAPIClientKey',      #
++    'StripePublishableKey',
+  ]
+}
+```
+
+---
+
+#### 2. Configure the library to consume our new key
+
+We'll need to update the `initWithUserId...` function -- one **fun** part of adjusting to Objective-C is that rather than named funtions, we just refer to them by their entire signatures -- to expose the new key as a property and add it to `constantsToExport`.
```

Comment:
`constantsToExport` could be a link to the docs: https://facebook.github.io/react-native/docs/native-modules-ios.html#exporting-constants